src/Nethermind/Nethermind.Init/Steps/InitializeNetwork.cs

```diff
@@ -214,8 +227,9 @@ private async Task Initialize(CancellationToken cancellationToken)
         }
         else if (_logger.IsDebug) _logger.Debug("Skipped enabling eth67 & eth68 capabilities");
 
-        if (_syncConfig.SnapSync && !stateSyncFinished)
+        if (_syncConfig.SnapSync)
         {
+            // TODO: Should we keep snap capability even after finishing sync?
```

Comment:
@marcindsobczak we now need snap protocol to heal potentially at any time

Reply:
This is needed only for the node which is corrupted, enabling it for all netherminds is overkill. Other nethermind nodes are useless anyway, as can't respond with snap data. We shouldn't keep not really supported capability for a longer time than absolutely necessary (so when we are syncing). It can bring potential problems with peering if we receive snap data requests from synced nodes (eg. recovering netherminds). Another thing, it will slow down syncing nodes, as they will connect with us, send snap data request and got disconnected. It is not a thing which we MUST avoid, but I think we can do it better.

Few better things than keeping snap protocol enabled all the time:
1) P2P protocol has a message `AddCapability`. On corrupted node, we can enable snap protocol and send that request to all peers. Final result will be even better, because we will not have snap protocol connection with other netherminds. After recovery, just disable snap cap, just like after sync. (but I never used this message and I'm not 100% sure that it is working this way - we need to check it)
2) Remove logic from eth67 protocol handler, which is ignoring `GetNodeData` and `NodeData` requests. When using eth67 protocol (or higher) we will not send such messages, but we will still respond to it. This way we will be able to use node data messages with all netherminds. Disadvantage of this solution is that it will be hardcoded hack and if other clients would like to use node data too, it will not be possible without changes in the code. And it will break eth protocol, but in the way not affecting other clients, so I think it is acceptable (but still not elegant)
3) We can add already implemented (but not tested at all as far) satellite protocol `NodeDataProtocol` and enable on all netherminds. If other clients would like to exchange node data, they can just implement this simple protocol. IMO it will be the cleanest and most straight-forward solution

IMO combination of 1 and 3 will be the best solution here. We could add logic for 1 now (instead of keeping snap enabled all the time) and add 3 when ready.

Reply:
In `ProtocolsManager` we have old method from 2019 `SendNewCapability` so looks like it should work this way. So after finding db corruption we would need to:
1) check if we have snap cap already enabled
2) if no, enable it
3) loop all peers and use `SendNewCapability` to add snap cap OR do it in healing classes, after checking snap cap - if is not available, try to add it. So instead of sending it to all peers, send only to the selected one
4) after recovery, disable snap cap